cmake/SociConfig.cmake

```diff
@@ -69,7 +69,7 @@ else()
             set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
         elseif (CMAKE_SYSTEM_NAME MATCHES "Windows")
             # We need to disable the warnings for using %I64d with MinGW.
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pedantic-ms-format")
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pedantic-ms-format -Wno-format")
```

Comment:
Are we sure the option is getting used there? As usual, I have no idea how to do anything with CMake, but perhaps there is a way to show the actual compilation lines with it?

If it is used, then the only explanation I have is that there was a bug in just 4.8 MinGW-W64 (as I have both 4.7 and 4.9 and they both work fine), but this would be pretty strange.